includes/class-validate-by-domain-options.php

```diff
@@ -0,0 +1,159 @@
+<?php
+/**
+ * All of the functionality for the plugins options page in the dashboard
+ */
+
+class Validate_By_Domain_Options {
+
+	public function plugin_admin_add_page() {
+
+		add_options_page( 'Validate by Domain', 'Validate by Domain', 'manage_options', 'validate_by_domain', [
+			$this,
+			'plugin_options_page'
+		] );
+
+	}
+
+	/**
+	 * Register the plugin settings, create fields
+	 */
+	function plugin_settings_init() {
+
+		$args = array(
+			'sanitize_callback' => [ $this, 'sanitize_input' ],
+		);
+
+		register_setting( 'validate_by_domain', 'validate_by_domain_settings', $args );
+
+		add_settings_section(
+			'options_section',
+			__( '', 'WordPress' ),
+			'',
+			'validate_by_domain'
+		);
+
+		add_settings_field(
+			'validate_enable',
+			__( '', 'WordPress' ),
+			[ $this, 'enable_render' ],
+			'validate_by_domain',
+			'options_section'
+		);
+
+		add_settings_field(
+			'validate_role',
+			__( '', 'WordPress' ),
+			[ $this, 'role_render' ],
+			'validate_by_domain',
+			'options_section'
+		);
+
+		add_settings_field(
+			'validate_whitelist',
+			__( 'Whitelist', 'WordPress' ),
+			[ $this, 'whitelist_render' ],
+			'validate_by_domain',
+			'options_section'
+		);
+
+	}
+
+// todo: Check for the existence of the key before using it in a condition, in the render functions below (avoid Illegal string offset)
+
+	/**
+	 * Render the options page enable field
+	 */
+	function enable_render() {
+
+		$options = get_option( 'validate_by_domain_settings' );
+		?><!-- <input
+        type='checkbox'
+        name='validate_by_domain_settings[validate_enable]' <?php checked( $options['validate_enable'], 1 ); ?>
+        value='1'>--><?php
+	}
+
+	/**
+	 * Render the options page role selection field
+	 */
+	function role_render() {
+
+		$options = get_option( 'validate_by_domain_settings' );
+		?>
+        <!-- <select name='validate_by_domain_settings[validate_role]'>
+            <option value='1' <?php selected( $options['validate_role'], 1 ); ?>>Contributor</option>
```

Comment:
php needs to be commented out with `//` - generating undefined index notices

Reply:
Done